### PR TITLE
[IMP] mrp: operations and workcenters views

### PR DESCRIPTION
**Pull request by svs-odoo**
Some changes/improvements in the list/form views of `mrp.workcenter` and `mrp.routing.workcenter`.

task-2382256

### CODE REVIEW

File: addons/mrp/models/mrp_routing.py [hunk 1]
```diff
@@ -7,14 +7,11 @@
 class MrpRoutingWorkcenter(models.Model):
     _name = 'mrp.routing.workcenter'
     _description = 'Work Center Usage'
-    _order = 'sequence, id'
+    _order = 'workcenter_id, bom_id'
     _check_company_auto = True
 
     name = fields.Char('Operation', required=True)
     workcenter_id = fields.Many2one('mrp.workcenter', 'Work Center', required=True, check_company=True)
-    sequence = fields.Integer(
```
Comment (Whenrow) [inline addons/mrp/models/mrp_routing.py:15]: Remove only from the view

**Review by ryv-odoo — APPROVED** [review]
LGTM!

File: addons/mrp/views/mrp_routing_views.xml [hunk 2]
```diff
@@ -75,6 +79,21 @@
...
         </record>
 
+        <record id="mrp_routing_workcenter_filter" model="ir.ui.view">
+            <field name="name">mrp.routing.workcenter.filter</field>
+            <field name="model">mrp.routing.workcenter</field>
+            <field name="arch" type="xml">
+                <search string="Operations Search Filters">
+                    <field name="bom_id"/>
+                    <field name="workcenter_id"/>
+                <group>
```
Comment (ryv-odoo) [inline addons/mrp/views/mrp_routing_views.xml:89]: Weird indent.

File: addons/mrp/views/mrp_workcenter_views.xml [hunk 3]
```diff
@@ -32,10 +32,18 @@
...
-                    <field name="code"/>
                     <field name="name"/>
+                    <field name="code"/>
+                    <field name="alternative_workcenter_ids" widget="many2many_tags" optional="hide"/>
+                    <field name="productive_time" optional="hide"/>
+                    <field name="costs_hour"/>
+                    <field name="capacity"/>
+                    <field name="time_efficiency" optional="hide"/>
+                    <field name="oee_target" optional="hide"/>
+                    <field name="time_start" optional="hide" string="Setup Time"/>
```
Comment (ryv-odoo) [inline addons/mrp/views/mrp_workcenter_views.xml:45]: Should be change in the model instead to be consistent with the form view? (same for time_stop)

File: addons/mrp_account/views/mrp_workcenter_views.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<odoo>
+    <record id="mrp_workcenter_tree_view_inherited" model="ir.ui.view">
+        <field name="name">mrp.workcenter.tree.inherited</field>
+        <field name="model">mrp.workcenter</field>
+        <field name="inherit_id" ref="mrp.mrp_workcenter_tree_view" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='costs_hour']" position="after">
+                <field name="costs_hour_account_id" optional="hide"/>
```
Comment (ryv-odoo) [inline addons/mrp_account/views/mrp_workcenter_views.xml:8]: For the form view, this fields is only added in enterprise (mrp_account_enterprise). Maybe change move from enterprise to community for the form too ?

**Review by ryv-odoo — COMMENTED** [review]
Some remarks.

File: addons/mrp_account/views/mrp_workcenter_views.xml (hunk 4 above)
Comment (svs-odoo) [reply addons/mrp_account/views/mrp_workcenter_views.xml]: I moved this part in enterprise.

File: addons/mrp/views/mrp_routing_views.xml [hunk 5]
```diff
@@ -6,11 +6,15 @@
             <field name="name">mrp.routing.workcenter.tree</field>
             <field name="model">mrp.routing.workcenter</field>
             <field name="arch" type="xml">
-                <tree string="Routing Work Centers">
-                    <field name="sequence" widget="handle"/>
+                <tree string="Routing Work Centers" multi_edit="1">
+                    <field name="sequence" widget="handle" attrs="{'column_invisible': [('parent.id', '=', False)]}"/>
                     <field name="name"/>
+                    <field name="bom_id"/>
```
Comment (ryv-odoo) [inline addons/mrp/views/mrp_routing_views.xml:12]: Modify this tree view, modify the `operations` tree view of the BoM form. Then I think we should **not** show the `bom_id` in that case, maybe with the `bom_id_invisible`  from context (like in form) + column_invisible?

**Comment by ryv-odoo** [comment]
@robodoo r+

**Comment by ryv-odoo** [comment]
@robodoo rebase ff

**Comment by ryv-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**